### ISSUE TITLE
Problem with Waterfox 2020.2 and GlibC 2.31

### ISSUE THREAD

**Issue by alebersol** [MrAlex94/Waterfox#1446]
Dear Mr. Kontos, There's an error with Waterfox 2020.2 and GlibC 2.31.

No sound in YouTube or any other streaming site. Either pulseaudio or Alsa. Ok ? Error: Sandbox: seccomp sandbox violation: pid 10245, tid 20082, syscall 220, args 5570560 139994008386436 2 0 8 55.

The numbers may vary, but get this error when try to play You Tube or any streaming site (either video or audio).

Regards,

**Comment by hawkeye116477** [comment]
Probably related => https://bugzilla.mozilla.org/show_bug.cgi?id=1600574

**Comment by hawkeye116477** [comment]
I have glibc 2.31 and with pulseaudio all is fine.

**Comment by steanne** [comment]
is this waterfox classic or current?

i have glibc 2.31 and sound is working for me in classic 2020.02.  i have not yet updated current to 2020.02.

**Comment by hawkeye116477** [comment]
For me, on both - Current and Classic sound works.

There is only problem with Widevine DRM (plugin crashes).

**Comment by alebersol** [comment]
My sound only works with that workaround of MOZ_DISABLE_CONTENT_SANDBOX=1

Let's hope it gets fixed soon.

Regards,

Em dom., 1 de mar. de 2020 às 09:41, hawkeye116477 <notifications@github.com>
escreveu:

**Comment by hawkeye116477** [comment]
Even set `media.cubeb.backend` to `alsa` and still works good.

**Comment by sysdbugfactory** [comment]
I can attest of this issue. several reports exists on reddit, arch linux bbs and manjaro support forum.

I am among the people affected by this issue which initally affected a number of web browsers but it seems waterfox is now the only browser that has not fixed it yet.

-edit- 
it may have been fixed with the recent release of Waterfox 2020.03, though I would not risk installing it at the moment due a number of report on waterfox dedicated subreddit warning of issues with some video not playing anymore, some webpages not loading correctly, or even crashes/lock up.
-/edit-

upstream firefox seems to have fixed the issue 3 days after it was reported according to the pclinuxos forum post which popularized the workaround of disabling the sandbox:
https://www.pclinuxos.com/forum/index.php/topic,151574.msg1295127.html#msg1295127

**Comment by hawkeye116477** [comment]
`it may have been fixed with the recent release of Waterfox 2020.03, though I would not risk installing it at the moment due a number of report on waterfox dedicated subreddit warning of issues with some video not playing anymore, some webpages not loading correctly, or even crashes/lock up.`

There is newer version 2020.03.1. Keep in mind that you can always revert to old version.

`some webpages not loading correctly`
This is nothing new, Classic version is based on FF 56 and some webmasters started using more advanced things...

**Comment by sysdbugfactory** [comment]
Sorry for the confusion I lacked precision in my edit

2020.03.1 is the version affected by the reported issues on the subreddit such as 
"Waterfox 2020.3.1 (64bit) does not play video or not show certain web pages"
"Waterfox classic 2020.03.1 crashes after allow microphone"
"Waterfox 2020.03.1 freezes within 3 minutes of loading even in safe mode!"
"Waterfox 2020.0301 (64 bit) Fails To Display Web Pages Correctly"

The webpages not loading correctly in 2020.03.x that were loading correctly in 2020.02, this is a newly introduced issue with the 2020.03.x release.

I'll wait until this settles down before updating and report back to tell if the sound issue is indeed fixed.

**Comment by sysdbugfactory** [comment]
Just updated to waterfox 2020.04-1, still no sound.

when trying to play sound with waterfox I get this error:

Sandbox: seccomp sandbox violation: pid 537659, tid 538192, syscall 220, args 18546745 139956337990844 2 0 6 139956449935456.

**Comment by sysdbugfactory** [comment]
here is the relevant bug report from arch linux:
https://bugs.archlinux.org/task/65469?order2=status&sort2=desc

here is the relevant bugzilla report from mozilla:
https://bugzilla.mozilla.org/show_bug.cgi?id=1600574

**Comment by hawkeye116477** [comment]
`here is the relevant bug report from arch linux`

That's report for Widevine crashing and that was already fixed.

**Comment by sysdbugfactory** [comment]
When searching about "Error: Sandbox: seccomp sandbox violation: ..."in relation to having no sound in either firefox or waterfox the main lead is this very bug reports, as both issues of lack of sound and brokem netflix DRM seem to both have the cause.

I'd argue it is relevant to the matter at hands despite your attempt to brush it off with the back of your hand, because it contains information about the syscall change that caused the whole ordeal.

The only other lead I found comes arch linux bbs and points to this old bug report mentioning  "seccomp sandbox violation" errors giving no sound  when using ALSA with firefox 58:
https://bugzilla.mozilla.org/show_bug.cgi?id=1430274
But this seems unrelated to the matter here.

Again fixing the widevine plugin in firefox alsa fixed the no sound error, I don't know about widevine status in waterfox as I do not use it, but I can attest the "no sound" issue is still here.

If the widevine plugin is fixed in waterfox, then what could be causing this fix not fixing the sound issue at the same time ?

Is there some kind of data can I provide to help debug this as I can reproduce the issue 100% ?

**Comment by alebersol** [comment]
Well, sound returned in the most recent version, in PCLinuxOS, without the
sandbox hack.

Best regards,



Em ter., 21 de abr. de 2020 às 07:03, sysdbugfactory <
notifications@github.com> escreveu:

**Comment by sysdbugfactory** [comment]
Good for them but how exactly is that helping those still affected by the issue ?

Providing context and details would help. What version exactly this the most recent version you are mentioning ? is it waterfox current or waterfox classic ? do they use pulseaudio or alsa ? 

Distrowatch points that the "most recent version" available on pclinuxos is waterfox-2020.03.1-1pclos2020.x86_64.rpm 
I am not using the most recent version available because it is not yet available on my distro but it is already more recent than the one avaialable on pclinuxos: the version I am using is waterfox-classic-bin 2020.04-1.

This rules out the lead that upgrading waterfox would fix the issue.

**Comment by alebersol** [comment]
Classic Waterfox 2020.03.1, with just ALSA, no pulseaudio.

Ok ?

Best regards,

Em qua., 22 de abr. de 2020 às 04:32, sysdbugfactory <
notifications@github.com> escreveu:

**Comment by alebersol** [comment]
Update: That worked with an LXDE installation. With a Trinity installation,
I have to resort to the sandbox hack (sadly).

Regards,



Em qua., 22 de abr. de 2020 às 08:49, Alessandro Ebersol <
alebersol@gmail.com> escreveu:

**Comment by sysdbugfactory** [comment]
Thanks for the details you provided. 

So we're both on the classic line, with alsa only and no pulseaudio which sets aside the issue would affect alsa only but not those using pulseaudio.

Could it be that pclinuxos provides a workaround at launch time by disabling sandbox ?
The other obvious explanation for the bug not affecting pclinuxos anymore could be glibc related.

On my machines installed glibc version is 2.31-2

**Comment by alebersol** [comment]
2.31-5 here.

But, I have to confess: It still needs the sandbox hack, waterfox classic,
or else, no sound at all.

Best regards,




Em sáb., 25 de abr. de 2020 às 12:19, sysdbugfactory <
notifications@github.com> escreveu:

**Comment by hawkeye116477** [comment]
You can try with mozregression find pushlog with fixes => https://aur.archlinux.org/packages/mozregression-gui. Just run a new bisection and choose last known bad and first known good build, then when you found last bad and first good, copy results with link for pushlog.

**Comment by sysdbugfactory** [comment]
Just reporting that I fixed the no sound issue in my waterfox installation after having to fix another issue.

TL;DR: Refreshing waterfox with the command `waterfox-classic -safe-mode` and selecting the refresh option was enough to return sound during video playback.

Here's the detailed account of what happened.

I just hit a serious issue where the address bar disappeared out of the blue after a high cpu load, the 3-bar menu would not appear and other functionality were not working. Despite closing the waterfox process would remain and use a lot of cpu cycles. I eventually stopped it with a sigterm signal. 

I did a backup copy of my user profile.

On restarting I was greeted with a popup from an extension warning me it was disabled and this could prevent another extension from working properly, the address bar was still gone, the previous session was not restored and many functionality still did not work.

I tried starting waterfox in safe mode with the follwing command: waterfox-classic -safe-mode
The first time I chose the option to start in safe mode, and confirmed that in safe mode waterfox worked as intended.
I started waterfox in safe mode a second time and chose to refresh it.

After the refresh, the address bar was back, the broken functionality are working again at the price of having to redo a number of settings and parameters. This also fixed my issue with having no sound in video pointing to the issue was coming from my installation, maybe a stale setting or some similar issue.

Hoping this could help  someone else.
